### PR TITLE
New Resource: azurerm_cosmosdb_sql_role_definition

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new resource `azurerm_cosmosdb_sql_role_definition`.

--- PASS: TestAccCosmosDbSQLRoleDefinition_complete (1365.53s)
--- PASS: TestAccCosmosDbSQLRoleDefinition_basic (1369.44s)
--- PASS: TestAccCosmosDbSQLRoleDefinition_requiresImport (1509.18s)
--- PASS: TestAccCosmosDbSQLRoleDefinition_update (1657.18s)

Rest API Spec:
https://github.com/Azure/azure-rest-api-specs/blob/184b23b63ee3a79fb41ca3aa55a31a8d9e636772/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-10-15/rbac.json#L79

### CODE REVIEW

File: website/docs/r/cosmosdb_sql_role_definition.html.markdown [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The GUID as the name of the Cosmos DB SQL Role Definition. Changing this forces a new resource to be created.
+
```
Comment (aristosvo) [inline website/docs/r/cosmosdb_sql_role_definition.html.markdown:58]: Hi @neil-yechenwei! Thanks, looking forward to seeing this implemented and released! I looked over the code quickly, will circle back as soon as I've a bit more time on my hands.

For now my main question is about naming in comparison to `azurerm_role_definition`. I looked at [`azurerm_role_definition` as a comparable resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_definition#argument-reference) and was wondering if it is better to build `azurerm_cosmos_sql_role_definition` in a similar way.

`azurerm_role_definition.name` is in that case comparable to `azurerm_cosmos_sql_role_definition.role_name`, and `azurerm_role_definition.role_definition_id` to `azurerm_cosmos_sql_role_definition.name`.

Does it make sense to use it in a similar way here, or is this something different? From a user perspective this is a bit harder to comprehend and confusing if these resources work differently.

Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_sql_role_definition.html.markdown]: Thanks for suggestion. I've updated code.

File: internal/services/cosmos/cosmosdb_sql_role_definition_resource.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"permissions": {
+				Type:     pluginsdk.TypeSet,
+				Required: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"data_actions": {
```
Comment (katbyte) [inline internal/services/cosmos/cosmosdb_sql_role_definition_resource.go:89]: Is this a block for a reason? or would it make more sense to pop this up a level into lets say `data_actions_allowed` ?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei - overall looks good i just had 1 question about schema i've left inline

File: internal/services/cosmos/cosmosdb_sql_role_definition_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_sql_role_definition_resource.go:89]: @katbyte , there is an existing resource [azurerm_role_definition](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_definition#permissions) that is similar with this RP. So as @aristosvo above mentioned, it is better to build this cosmosdb sql role definition in a similar way. Then user could be easier to use and understand.

**Review by katbyte — APPROVED** [review]
LGTM then @neil-yechenwei 🦀

**Pull request merged**